### PR TITLE
feat: added support for cobadged cards

### PR DESCRIPTION
**Pull request by Sanskar2001**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Added support to process co badged cards

## How did you test it?
Tested it locally on sandbox

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/CardSchemeComponent.res [hunk 1]
```diff
@@ -0,0 +1,37 @@
+module CoBadgeCardSchemeDropDown = {
```
Comment (PritishBudhiraja) [inline src/CardSchemeComponent.res:1]: We have our drop down component cam we use that only?

File: src/CardSchemeComponent.res [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  @react.component
+  let make = (~cardNumber, ~setCardBrand) => {
+    <select
+      className="w-[15px]"
```
Comment (PritishBudhiraja) [inline src/CardSchemeComponent.res:5]: can we give in rem value and based on tailwind provided values?

File: src/CardSchemeComponent.res [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      onChange={ev => {
+        let target = ev->ReactEvent.Form.target
+        let value = target["value"]
+        setCardBrand(_ => value)
+      }}>
+      <option disabled=true> {"Select a card brand"->React.string} </option>
```
Comment (PritishBudhiraja) [inline src/CardSchemeComponent.res:11]: why this option? It should be a label

File: src/CardSchemeComponent.res [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      {cardNumber
+      ->CardUtils.getCoBadgesCardSchemes
+      ->Array.mapWithIndex((item: string, i) => {
```
Comment (PritishBudhiraja) [inline src/CardSchemeComponent.res:14]: explicit type not required

File: src/CardSchemeComponent.res [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  let cardType = React.useMemo1(_ => cardBrand->CardUtils.getCardType, [cardBrand])
+  let animate = cardType == NOTFOUND ? "animate-slideLeft" : "animate-slideRight"
+
+  let cardBrandIcon = React.useMemo1(
+    _ => CardUtils.getCardBrandIcon(cardType, paymentType),
+    [cardBrand],
+  )
+  <div className={`${animate} flex items-center`}>
+    cardBrandIcon
+    {isCardCoBadged ? <CoBadgeCardSchemeDropDown cardNumber setCardBrand /> : React.null}
```
Comment (PritishBudhiraja) [inline src/CardSchemeComponent.res:35]: RenderIf component can be used

File: src/Payment.res [hunk 6]
```diff
@@ -47,13 +47,15 @@ let make = (~paymentMode, ~integrateError, ~logger) => {
...
-    !showFields && isNotBancontact ? (cardScheme, maxLength) : (brand, maxLength)
+  let maxCardLength = React.useMemo(() => {
+    getMaxLength(cardNumber)
   }, (cardNumber, cardScheme, showFields))
 
+  let cardBrand = getCardBrand(cardNumber)
+  let isNotBancontact = selectedOption !== "bancontact_card" && cardBrand == ""
+  let (cardBrand, setCardBrand) = React.useState(_ =>
+    !showFields && isNotBancontact ? cardScheme : cardBrand
+  )
```
Comment (PritishBudhiraja) [inline src/Payment.res:58]:
```suggestion
  let val = !showFields && isNotBancontact ? cardScheme : cardBrand
  let (cardBrand, setCardBrand) = React.useState(_ => val)
```
Define this outside

File: src/Payment.res [hunk 7]
```diff
@@ -377,11 +379,15 @@ let make = (~paymentMode, ~integrateError, ~logger) => {
     None
   }, (isExpiryValid, isExpiryComplete(cardExpiry)))
 
+  React.useEffect(() => {
+    // Console.log("changing card brand" ++ cardBrand ++ cardNumber)
```
Comment (PritishBudhiraja) [inline src/Payment.res:383]: remove not req comments

File: src/Components/PaymentInputField.res [hunk 8]
```diff
@@ -20,6 +20,7 @@ let make = (
   ~placeholder="",
   ~appearance: CardThemeType.appearance,
   ~className="",
+  ~rightIconClassName="-ml-10",
```
Comment (PritishBudhiraja) [inline src/Components/PaymentInputField.res:23]: one place it's -ml-10 another place -ml-12

Comment (Sanskar2001) [reply src/Components/PaymentInputField.res:23]: -ml-10 is the standard size applied in every input field for icon, however the card scheme component with cobadge component requires some extra margin as it is bigger in width.

File: src/CardSchemeComponent.res (hunk 1 above)
Comment (Sanskar2001) [reply src/CardSchemeComponent.res:1]: This drop down requires some additional customisation like the label (select a card brand) should be shown in the dropdown list and not as a hint in select, only the dropdown arrow (chevron) should be visible while selecting. Exposing additional props to achieve things like these could have made our reusable component messy. As these requirements are unique to this co badge drop down. Now this card scheme component can be reused anywhere in the SDK wherever the user is required to select the card scheme.

File: src/CardSchemeComponent.res (hunk 3 above)
Comment (Sanskar2001) [reply src/CardSchemeComponent.res:11]: We want to show in this inside dropdown list and not as a hint in select

**Pull request merged**